### DISCUSSION TITLE
OFDM resource grid with different CP sizes (per symbol) simulated in time-domain

### DISCUSSION THREAD

**Discussion by lidorsofer** [NVlabs/sionna#185] [Q&A]
Hello,

Is there currently an option to generate an OFDM grid with different CP size per symbol, convolved with time-domain channel model (e.g. TDLA)?

Thank you!
Lidor

**Comment by jhoydis** [comment]
Hi Lidor,

The [``OFDM.Modulator``](https://nvlabs.github.io/sionna/api/ofdm.html#sionna.ofdm.OFDMModulator) allows you to set the cyclic prefix as parameter. You can then use [``TimeChannel``](https://nvlabs.github.io/sionna/api/channel.wireless.html#timechannel) initialised with some channel model such as [``TDL``](https://nvlabs.github.io/sionna/api/channel.wireless.html#tapped-delay-line-tdl) to generate the channel outputs.

If you want to have a different CP length for every symbol in a sequence of OFDM symbols you would need to manually create the channel input by concatenating the outputs of different  [``OFDM.Modulator``](https://nvlabs.github.io/sionna/api/ofdm.html#sionna.ofdm.OFDMModulator) objects.

Hope that helps.
